src/lib/Layout/SideBarLayout/SideBarLayoutContext.tsx

```diff
@@ -1,6 +1,7 @@
 import { createContext, PropsWithChildren, ReactNode, useContext } from "react";
 
 interface SideBarLayoutProviderProps {
+  theme: "dark" | "light";
```

Comment:
Thinking about this again:

It's actually bad design to use the same interface for the `SideBarLayoutProvider` and for the `SideBarLayoutContext`. 

IMO the `SideBarLayoutProvider` should have the optional theme but the `SideBarLayoutContext` should have a required theme because we set a default here. 

Than in the `SideBarLayout` you wouldn't have to add another default because you already know that `theme` has a value for sure. 

So I would suggest this:

```
interface SideBarLayoutProviderProps {
  theme?: "dark" | "light";
  footer?: ReactNode;
  brand?: ReactNode;
  userDropDownMenuToggle?: ReactNode;
  userDropDownMenu?: ReactNode;
}

interface SideBarLayoutContextProps extends SideBarLayoutProviderProps {
  theme: "dark" | "light";
}

const SideBarLayoutContext = createContext<SideBarLayoutContextProps | null>(null);
```

You can also be more fancy with TypeScirpt utility types to construct this `SideBarLayoutContextProps` but for now this gets the job done quite well.

Reply:
implemented as suggested

Reply:
It's only not a breaking change if I add a default here. Hope I did it correctly. Not sure if I had to change the SidebarLayoutProvider to use ContextProps or ProviderProps. I think it makes sense the way it is now. otherwise let me know.